### PR TITLE
Fixes #13740 - Sets the default pulp-admin password

### PR DESCRIPTION
**Pull request by parthaa**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -40,6 +40,8 @@
 # $enable_ostree::      Boolean to enable ostree plugin. This requires existence of an ostree install.
 #                       type:boolean
 #
+# $pulp_admin_password:: password for the Pulp admin user. Leave blank for  random password to be generated
```
Comment (ekohl) [inline manifests/init.pp]: I think this should be 'Defaults to a random password' instead of leave blank.

File: manifests/params.pp [hunk 2]
```diff
@@ -67,4 +67,5 @@
   $qpid_url = "amqp:ssl:${::fqdn}:5671"
   $candlepin_event_queue = 'katello_event_queue'
   $enable_ostree = false
+  $pulp_admin_password   = cache_data('foreman_cache_data', 'pulp_node_admin_password', random_password(32))
```
Comment (ekohl) [inline manifests/params.pp]: Double spaces before =.

**Comment by ekohl** [comment]
Small nitpicks, otherwise ACK

**Comment by parthaa** [comment]
@ekohl @ehelms updated

File: manifests/params.pp [hunk 3]
```diff
@@ -68,4 +68,5 @@
   $candlepin_event_queue = 'katello_event_queue'
   $enable_ostree = false
   $max_keep_alive = 10000
+  $pulp_admin_password   = cache_data('foreman_cache_data', 'pulp_node_admin_password', random_password(32))
```
Comment (ekohl) [inline manifests/params.pp]: Still odd spacing before =

**Comment by parthaa** [comment]
@ekohl updated

**Comment by ekohl** [comment]
ACK

**Comment by ekohl** [comment]
Needs a rebase

**Comment by ehelms** [comment]
@parthaa do you still want this functionality? If you rebase it we can get it in. Otherwise let us know and we'll close it.

File: manifests/params.pp [hunk 4]
```diff
@@ -68,4 +68,5 @@
   $candlepin_event_queue = 'katello_event_queue'
   $enable_ostree = false
   $max_keep_alive = 10000
+  $pulp_admin_password = cache_data('foreman_cache_data', 'pulp_node_admin_password', random_password(32))
```
Comment (ehelms) [inline manifests/params.pp]: We should drop the 'node' notation here.

**Comment by ehelms** [comment]
@parthaa want to revisit this? We can get it in this week. Otherwise I'll give it a week and close it out to be re-opened later.

**Comment by ekohl** [comment]
Should this be revisited or be closed?

**Comment by ekohl** [comment]
Closing for inactivity.

**Pull request closed without merging**